### PR TITLE
[BUG] fix tag errors, refactor comments np.array -> np.ndarray

### PR DESCRIPTION
**Pull request by TonyBagnall**
whilst removing datatypes references, I noticed a tag in y_type  

```python
    (
        "y_inner_type",
        ["forecaster", "transformer"],
        (
            "list",
            [
                "pd.Series",
                "pd.DataFrame",
                "np.array",
                "nested_univ",
                "pd-multiindex",
                "numpy3D",
                "df-list",
            ],
        ),
```
this is not a valid type even in the complete madness of datatypes, since this

```
x = mtype_to_scitype("np.array")
```
gives an error ValueError: np.array is not a supported mtype

Also noticed many forecasters assign     "pd_multiindex_hier" to y_inner_type, so I have also added that to the tag registry. 

I then look further and see many references in comments to np.array, which as I understand it is a function to create an array, whereas the type should be np.ndarray, so I have refactored them,.

X_inner_types also had np.array tag, and also was listed for "forecaster" only, when in truth this tag is used in all collection estimators

### CODE REVIEW

File: aeon/classification/_delegate.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def _fit(self, X, y):
 
         Parameters
         ----------
-        X : 3D np.array
+        X : 3D np.ndarray
```
Comment (chrisholder) [inline aeon/classification/_delegate.py:50]: Added shape of np.ndarray in other docstring maybe add in all these ones with just 3D np.ndarray for consistency

File: aeon/classification/_dummy.py [hunk 2]
```diff
@@ -83,7 +83,7 @@ def _fit(self, X, y):
 
         Parameters
         ----------
-        X : 3D np.array of shape = [n_instances, n_channels, series_length]
+        X : 3D np.ndarray of shape = [n_instances, n_channels, series_length]
         y : array-like, shape = [n_instances] - the class labels
```
Comment (chrisholder) [inline aeon/classification/_dummy.py:87]: Inconsistent docstring  3D np.ndarray of shape = [n_instances, n_channels, series_length].

Make it consistent with classification:

X : 3D np.array
    Input data, any number of channels, equal length series of shape ``(
    n_instances, n_channels, n_timepoints)``
    or 2D np.array (univariate, equal length series) of shape
    ``(n_instances, n_timepoints)``

File: aeon/classification/base.py [hunk 3]
```diff
@@ -227,7 +227,7 @@ def score(self, X, y) -> float:
 
         Parameters
         ----------
-        X : 3D np.array (any number of channels, equal length series)
+        X : 3D np.ndarray (any number of channels, equal length series)
```
Comment (chrisholder) [inline aeon/classification/base.py:230]: Inconsistent docstring   X : 3D np.ndarray (any number of channels, equal length series)

Make it consistent with classification:

X : 3D np.array
    Input data, any number of channels, equal length series of shape ``(
    n_instances, n_channels, n_timepoints)``
    or 2D np.array (univariate, equal length series) of shape
    ``(n_instances, n_timepoints)``

**Review by chrisholder — CHANGES_REQUESTED** [review]
Need to make the docstring consistent with the other modules.

File: aeon/classification/_delegate.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/classification/_delegate.py:50]: I think it should be 
X : np.ndarray
  then desciption. 
  I think its meant to be just the type after the name

**Pull request merged**